specs/reference/data-services/models.v1.json

```diff
@@ -0,0 +1,141 @@
+{
+  "openapi": "3.0.0",
+  "info": {
+    "title": "models",
+    "version": "1.0",
+    "description": "Data services for endpoints pertaining to entity models"
+  },
+  "servers": [
+    {
+      "url": "http://localhost:8011/data-hub/5/data-services/models"
+    }
+  ],
+  "paths": {
+    "/getPrimaryEntityTypes.sjs": {
+      "post": {
+        "summary": "getPrimaryEntityTypes",
+        "tags": [],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "../../models/PrimaryEntityTypes.v1.json"
+                }
+              }
+            }
+          }
+        },
+        "operationId": "getPrimaryEntityTypes.sjs",
+        "description": "Returns an array of primary entity types. A primary entity type is the entity type in a model descriptor with a name equal to the title of the model descriptor."
+      },
+      "parameters": []
+    },
+    "/createModel.sjs": {
+      "post": {
+        "summary": "createModel",
+        "operationId": "post-createModelsjs",
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "application/json": {
```

Comment:
Do we want to change this to actually be a multipart request and use encoding to reference the JSON schema as part of the "input" parameter? Like this: https://swagger.io/docs/specification/describing-request-body/multipart-requests/ That would more accurately describe the request being made and could be eventually be leveraged for testing data services. Of course we have a good pattern for ML unit tests and Data Services, so that may be less important.

Reply:
Yeah I struggled with how to map to DS parameters. The fact that it's multi-part is an implementation detail that's hidden from developers too, buried away in the DS-generated code. So if we want to use OpenAPI to describe DS interfaces, I think we just use a combo of query params and request body as JSON.